### PR TITLE
fix: add password/secret length check to be <= 128

### PR DESCRIPTION
**Pull request by zyyw**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)
Resolve task 5 in issue 18091

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/controller/robot/controller.go [hunk 1]
```diff
@@ -416,5 +416,5 @@ var (
 )
 
 func IsValidSec(secret string) bool {
-	return len(secret) >= 8 && hasLower.MatchString(secret) && hasUpper.MatchString(secret) && hasNumber.MatchString(secret)
+	return len(secret) >= 8 && len(secret) <= 128 && hasLower.MatchString(secret) && hasUpper.MatchString(secret) && hasNumber.MatchString(secret)
```
Comment (wy65701436) [inline src/controller/robot/controller.go:419]: please add UI for this change.

File: src/server/v2.0/handler/user.go [hunk 2]
```diff
@@ -456,10 +456,10 @@ func requireValidSecret(in string) error {
 	hasLower := regexp.MustCompile(`[a-z]`)
 	hasUpper := regexp.MustCompile(`[A-Z]`)
 	hasNumber := regexp.MustCompile(`[0-9]`)
-	if len(in) >= 8 && hasLower.MatchString(in) && hasUpper.MatchString(in) && hasNumber.MatchString(in) {
+	if len(in) >= 8 && len(in) <= 128 && hasLower.MatchString(in) && hasUpper.MatchString(in) && hasNumber.MatchString(in) {
```
Comment (wy65701436) [inline src/server/v2.0/handler/user.go:459]: ditto

**Review by AllForNothing — APPROVED** [review]
lgtm

**Review by wy65701436 — APPROVED** [review]
Lgtm

**Pull request merged**